egs/librimix/ConvTasNet/eval.py

```diff
@@ -0,0 +1,146 @@
+                         '`sep_clean` or `sep_noisy`')
+parser.add_argument('--out_dir', type=str, required=True,
+                    help='Directory in exp_dir where the va will be stored')
+parser.add_argument('--use_gpu', type=int, default=0,
+                    help='Whether to use the GPU for model execution')
+parser.add_argument('--exp_dir', default='exp/tmp',
+                    help='Experiment root')
+parser.add_argument('--n_save_ex', type=int, default=10,
+                    help='Number of audio examples to save, -1 means all')
+
+compute_metrics = ['si_sdr', 'sdr', 'sir', 'sar', 'stoi']
+
+
+def main(conf):
+    # Make the model
+    model, _ = make_model_and_optimizer(conf['train_conf'])
+    # Load best model
+    with open(os.path.join(conf['exp_dir'], 'best_k_models.json'), "r") as f:
+        best_k = json.load(f)
+    best_model_path = min(best_k, key=best_k.get)
+    # Load checkpoint
+    checkpoint = torch.load(best_model_path, map_location='cpu')
+    state = checkpoint['state_dict']
+    state_copy = state.copy()
+    # Remove unwanted keys
+    for keys, values in state.items():
+        if keys.startswith('loss'):
+            del state_copy[keys]
+            print(keys)
+    model = load_state_dict_in(state_copy, model)
+    # Handle device placement
+    if conf['use_gpu']:
+        model.cuda()
+    model_device = next(model.parameters()).device
+    test_set = LibriMix(csv_dir=conf['test_dir'],
+                        task=conf['task'],
+                        sample_rate=conf['sample_rate'],
+                        n_src=conf['train_conf']['data']['n_src'],
+                        segment=None) # Uses all segment length
+    # Used to reorder sources only
+    loss_func = PITLossWrapper(pairwise_neg_sisdr, mode='pairwise')
+
+    # Randomly choose the indexes of sentences to save.
+    eval_save_dir = os.path.join(conf['exp_dir'], conf['out_dir'])
+    ex_save_dir = os.path.join(eval_save_dir, 'examples/')
+    if conf['n_save_ex'] == -1:
+        conf['n_save_ex'] = len(test_set)
+    save_idx = random.sample(range(len(test_set)), conf['n_save_ex'])
+    series_list = []
+    torch.no_grad().__enter__()
+    for idx in tqdm(range(len(test_set))):
+        # Forward the network on the mixture.
+        mix, sources = tensors_to_device(test_set[idx], device=model_device)
+        est_sources = model(mix.unsqueeze(0))
+        loss, reordered_sources = loss_func(est_sources, sources[None],
+                                            return_est=True)
+        mix_np = mix.cpu().data.numpy()
+        sources_np = sources.squeeze().cpu().data.numpy()
+        est_sources_np = reordered_sources.squeeze().cpu().data.numpy()
+        # Declipping the sources
+        est_sources_np[0, :] = est_sources_np[0, :] / \
+                               np.max(est_sources_np, axis=1)[0]
+        est_sources_np[1, :] = est_sources_np[1, :] / \
+                               np.max(est_sources_np, axis=1)[1]
```

Comment:
This is useful only before saving. 
All our metrics are invariant to scale here this won't always be the case. 
Unless there's a good reason, the normalization should be just before sf.write.

Reply:
Ok i'll make the change

Reply:
Also realized this syntax is bad. It only works  for 2 sources I will just remove it.